builtin/providers/google/provider.go

```diff
@@ -264,3 +265,63 @@ func handleNotFoundError(err error, d *schema.ResourceData, resource string) err
 
 	return fmt.Errorf("Error reading %s: %s", resource, err)
 }
+
+var regexpUrl = regexp.MustCompile(fmt.Sprintf("^(?:https://www.googleapis.com/compute/%s/)?(projects/%s(?:/(?:global|regions/%s|zones/%s)(?:/%s/%s)?)?)$",
+	"([a-z0-9]+)",                    // API Version
+	"([a-z](?:[-a-z0-9]*[a-z0-9])?)", // Project ID
+	"([a-z]+-[a-z]+[0-9]+)",          // Region
+	"([a-z]+-[a-z]+[0-9]+-[a-z]+)",   // Zone
+	"([a-z][A-Za-z]*(?:/family)?)",   // Resource type
+	"([a-z](?:[-a-z0-9]*[a-z0-9])?)", // Resource name
+))
+
+type resourceInfo struct {
+	apiVersion   string
+	project      string
+	region       string
+	zone         string
+	resourceType string
+	name         string
+	url          string
+}
+
+func parseUrl(url string) (resourceInfo, error) {
```

Comment:
This function is kinda over-spec only for this purpose, but hope it will be useful on later usage.